### ISSUE TITLE
Compile errors when instanciating a child process

### ISSUE THREAD

**Issue by hotgloupi** [klemens-morgenstern/boost-process#21]
Hi, I'm trying to compile a simple hello world using g++4.8 against boost 1.55:

```
#include <boost/process.hpp>

namespace bp = boost::process;

int main()
{
    std::string cmd = "echo 'Hello, world!'";

    bp::child child(
        cmd,
        bp::shell,
        bp::std_in.close(),
        bp::std_err.close(),
        bp::std_out.close()
    );

    child.wait();
    return child.exit_code();
}
```

This is what I get:

```
In file included from /boost-1.55/include/boost/fusion/container/vector/vector10.hpp:65:0,
                 from /boost-1.55/include/boost/fusion/container/vector/vector_fwd.hpp:15,
                 from /boost-1.55/include/boost/fusion/mpl/detail/clear.hpp:11,
                 from /boost-1.55/include/boost/fusion/mpl/clear.hpp:13,
                 from /boost-1.55/include/boost/fusion/mpl.hpp:17,
                 from /boost-1.55/include/boost/fusion/view/detail/strictest_traversal.hpp:16,
                 from /boost-1.55/include/boost/fusion/view/transform_view/transform_view.hpp:21,
                 from /boost-1.55/include/boost/fusion/algorithm/transformation/transform.hpp:11,
                 from src/boost/process/detail/posix/io_service_ref.hpp:16,
                 from src/boost/process/async.hpp:30,
                 from src/boost/process.hpp:21,
                 from test.cpp:1:
/boost-1.55/include/boost/fusion/container/vector/detail/preprocessed/vector10.hpp: In instantiation of ‘boost::fusion::vector_data1<T0>::vector_data1(typename boost::fusion::detail::call_param<Car>::type) [with T0 = boost::process::detail::posix::exe_cmd_init; typename boost::fusion::detail::call_param<Car>::type = const boost::process::detail::posix::exe_cmd_init&]’:
/boost-1.55/include/boost/fusion/container/vector/detail/preprocessed/vector10.hpp:54:36:   required from ‘static boost::fusion::vector_data1<T0> boost::fusion::vector_data1<T0>::init_from_sequence(const Sequence&) [with Sequence = boost::fusion::tuple<boost::process::detail::posix::exe_cmd_init, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_>; T0 = boost::process::detail::posix::exe_cmd_init]’
/boost-1.55/include/boost/fusion/container/vector/detail/preprocessed/vector10.hpp:128:58:   required from ‘boost::fusion::vector1<T0>::vector1(const Sequence&, typename boost::disable_if<boost::is_convertible<Sequence, Car> >::type*) [with Sequence = boost::fusion::tuple<boost::process::detail::posix::exe_cmd_init, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_>; T0 = boost::process::detail::posix::exe_cmd_init; typename boost::disable_if<boost::is_convertible<Sequence, Car> >::type = void]’
/boost-1.55/include/boost/fusion/container/vector/detail/preprocessed/vvector10.hpp:43:50:   required from ‘boost::fusion::vector<T0, T1, T2, T3, T4, T5, T6, T7, T8, T9>::vector(const Sequence&) [with Sequence = boost::fusion::tuple<boost::process::detail::posix::exe_cmd_init, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_, boost::fusion::void_>; T0 = boost::process::detail::posix::exe_cmd_init; T1 = boost::fusion::void_; T2 = boost::fusion::void_; T3 = boost::fusion::void_; T4 = boost::fusion::void_; T5 = boost::fusion::void_; T6 = boost::fusion::void_; T7 = boost::fusion::void_; T8 = boost::fusion::void_; T9 = boost::fusion::void_]’
/boost-1.55/include/boost/fusion/tuple/detail/preprocessed/tuple10.hpp:20:28:   required from ‘boost::fusion::tuple<T0, T1, T2, T3, T4, T5, T6, T7, T8, T9>::tuple(const boost::fusion::tuple<T0, T1, T2, T3, T4, T5, T6, T7, T8, T9>&) [with T0 = boost::process::detail::posix::exe_cmd_init; T1 = boost::fusion::void_; T2 = boost::fusion::void_; T3 = boost::fusion::void_; T4 = boost::fusion::void_; T5 = boost::fusion::void_; T6 = boost::fusion::void_; T7 = boost::fusion::void_; T8 = boost::fusion::void_; T9 = boost::fusion::void_]’
src/boost/process/detail/execute_impl.hpp:205:75:   required from ‘boost::process::child boost::process::detail::execute_impl(Args&& ...) [with Args = {std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const boost::process::detail::shell_&, boost::process::detail::posix::close_in, boost::process::detail::posix::close_out<2, -1>, boost::process::detail::posix::close_out<1, -1>}]’
src/boost/process/child.hpp:34:79:   required from ‘boost::process::child::child(Args&& ...) [with Args = {std::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const boost::process::detail::shell_&, boost::process::detail::posix::close_in, boost::process::detail::posix::close_out<2, -1>, boost::process::detail::posix::close_out<1, -1>}]’
test.cpp:15:5:   required from here
/boost-1.55/include/boost/fusion/container/vector/detail/preprocessed/vector10.hpp:34:20: error: use of deleted function ‘boost::process::detail::posix::exe_cmd_init::exe_cmd_init(const boost::process::detail::posix::exe_cmd_init&)’
             : m0(_0) {}
                    ^
In file included from src/boost/process/detail/basic_cmd.hpp:19:0,
                 from src/boost/process/args.hpp:20,
                 from src/boost/process.hpp:20,
                 from test.cpp:1:
src/boost/process/detail/posix/basic_cmd.hpp:104:5: error: declared here
     exe_cmd_init(const exe_cmd_init & ) = delete;
     ^
```

However, if I make `exe_cmd_init` copyable, it compiles fine and seems to work

**Comment by klemens-morgenstern** [comment]
For `exe_cmd_init` the deleted move-ctor is purely for performance. However: there are other initializers, where that might be not the case. I think gcc 4.8 doesn't have complete (or at least buggy) C++11 support. This together with #20 would be Gcc 4.8 workarounds, so I'll close this issue, so we'd have it in one thread.

Btw: thanks for trying the library!

**Comment by HeinrichGit** [comment]
I have the same issue with g++5.4 and Boost 1.56

**Comment by klemens-morgenstern** [comment]
I can build it with gcc-5.3 it is a boost.fusion issue, I guess fixed in 1.60. Boost.Process officially requires boost 1.63, without which you cannot use it windows. I hope it's going to ship in 1.64, so that the official dependency will be that version.
